### PR TITLE
Default Timeout

### PR DESCRIPTION
**Pull request by spape**
_MINOR VERSION_

## Timeout

Working and configuring timeouts is important, to ensure your app stays alive when services you depend on start to get really slow...

LHC forwards two timeout options directly to typhoeus:

`timeout` (in seconds) - Pass a long as parameter containing timeout - the maximum time in seconds that you allow the libcurl transfer operation to take. Normally, name lookups can take a considerable time and limiting operations to less than a few minutes risk aborting perfectly normal operations. This option may cause libcurl to use the SIGALRM signal to timeout system calls.
`connecttimeout` (in seconds) - Pass a long. It should contain the maximum time in seconds that you allow the connection phase to the server to take. This only limits the connection phase, it has no impact once it has connected. Set to zero to switch to the default built-in connection timeout - 300 seconds. 

```ruby
LHC.get('http://local.ch', timeout: 5, connecttimeout: 1)
```

LHC provides a [timeout interceptor](docs/interceptors/default_timeout.md) that let you apply default timeout values to all the requests that you are performig in your application.

# Default Timeout Interceptor

Applies default timeout values to all requests made in an application, that uses LHC.

```ruby
  LHC.config.interceptors = [LHC::DefaultTimeout]
```

`timeout` default: 15 seconds
`connecttimeout` default: 1 second

## Overwrite defaults

```ruby
LHC::DefaultTimeout.timeout = 5 # seconds
LHC::DefaultTimeout.connecttimeout = 2 # second
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,6 +112,21 @@ You can configure global endpoints, placeholders and interceptors.
 
 → [Read more about configuration](docs/configuration.md)
 
+## Timeout
+
+Working and configuring timeouts is important, to ensure your app stays alive when services you depend on start to get really slow...
+
+LHC forwards two timeout options directly to typhoeus:
+
+`timeout` (in seconds) - Pass a long as parameter containing timeout - the maximum time in seconds that you allow the libcurl transfer operation to take. Normally, name lookups can take a considerable time and limiting operations to less than a few minutes risk aborting perfectly normal operations. This option may cause libcurl to use the SIGALRM signal to timeout system calls.
```
Comment (crackofdusk) [inline README.md]: What do you mean by "pass a long"? A pass-through parameter?

File: README.md [hunk 2]
```diff
@@ -112,6 +112,21 @@ You can configure global endpoints, placeholders and interceptors.
...
+`connecttimeout` (in seconds) - Pass a long. It should contain the maximum time in seconds that you allow the connection phase to the server to take. This only limits the connection phase, it has no impact once it has connected. Set to zero to switch to the default built-in connection timeout - 300 seconds.
```
Comment (crackofdusk) [inline README.md]: I also don't understand the meaning of "pass a long" here.

File: README.md [hunk 3]
```diff
@@ -112,6 +112,21 @@ You can configure global endpoints, placeholders and interceptors.
...
+LHC forwards two timeout options directly to typhoeus:
+
+`timeout` (in seconds) - Pass a long as parameter containing timeout - the maximum time in seconds that you allow the libcurl transfer operation to take. Normally, name lookups can take a considerable time and limiting operations to less than a few minutes risk aborting perfectly normal operations. This option may cause libcurl to use the SIGALRM signal to timeout system calls.
+`connecttimeout` (in seconds) - Pass a long. It should contain the maximum time in seconds that you allow the connection phase to the server to take. This only limits the connection phase, it has no impact once it has connected. Set to zero to switch to the default built-in connection timeout - 300 seconds. 
+
+```ruby
+LHC.get('http://local.ch', timeout: 5, connecttimeout: 1)
+```
+
+LHC provides a [timeout interceptor](docs/interceptors/default_timeout.md) that let you apply default timeout values to all the requests that you are performig in your application.
```
Comment (crackofdusk) [inline README.md]: "that let**s**"

File: README.md (hunk 1 above)
Comment (Averethel) [inline README.md]: I believe you meant:

> Normally, name lookups can take a considerable time and limiting operations to less than a few **seconds** risk aborting perfectly normal operations

here

Comment (Averethel) [inline README.md]: > Pass a long as parameter

I would simply write `pass a number` instead. It's a bit more understandable this way.

File: README.md (hunk 2 above)
Comment (Averethel) [inline README.md]: > Pass a long

I would simply write `pass a number` instead. It's a bit more understandable this way.

File: README.md (hunk 3 above)
Comment (Averethel) [inline README.md]: > that **lets** you apply default

File: docs/interceptors/default_timeout.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+```
+
+`timeout` default: 15 seconds
+`connecttimeout` default: 1 second
+
+## Overwrite defaults
+
+```ruby
+LHC::DefaultTimeout.timeout = 5 # seconds
+LHC::DefaultTimeout.connecttimeout = 2 # second
```
Comment (Averethel) [inline docs/interceptors/default_timeout.md]: > **seconds**

File: README.md (hunk 1 above)
Comment (Averethel) [reply README.md]: My understanding is that it's  a `long` as in data type in C, but would still rephrase that

Comment (spape) [reply README.md]: Copy pasted from the C documentation of libcurl... deleted this part 👍

File: README.md (hunk 2 above)
Comment (spape) [reply README.md]: Copy pasted from the C documentation of libcurl... deleted this part 👍

File: README.md (hunk 1 above)
Comment (spape) [reply README.md]: 👍

Comment (spape) [reply README.md]: dropped that part.

Comment (spape) [reply README.md]: we have examples, too

File: README.md (hunk 2 above)
Comment (spape) [reply README.md]: dropped that part, we have examples too

File: README.md (hunk 3 above)
Comment (spape) [reply README.md]: 👍

File: docs/interceptors/default_timeout.md (hunk 4 above)
Comment (spape) [reply docs/interceptors/default_timeout.md]: 👍

**Pull request merged**